modules/components/pages/outputs/ockam_kafka.adoc

```diff
@@ -0,0 +1,706 @@
+| `OAUTHBEARER`
+| OAuth Bearer-based authentication.
+| `PLAIN`
+| Plain text authentication.
+| `SCRAM-SHA-256`
+| SCRAM-based authentication as specified in RFC5802.
+| `SCRAM-SHA-512`
+| SCRAM-based authentication as specified in RFC5802.
+| `none`
+| Disable SASL authentication
+|===
+
+=== `kafka.sasl[].username`
+
+A username for `PLAIN` or `SCRAM-*` authentication.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].password`
+
+A password for `PLAIN` or `SCRAM-*` authentication.
+
+include::components:partial$secret_warning.adoc[]
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].token`
+
+The token to use for a single session's `OAUTHBEARER` authentication.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].extensions`
+
+Key/value pairs to add to `OAUTHBEARER` authentication requests.
+
+*Type*: `object`
+
+=== `kafka.sasl[].aws`
+
+AWS specific fields for when the `mechanism` is set to `AWS_MSK_IAM`.
+
+*Type*: `object`
+
+=== `kafka.sasl[].aws.region`
+
+The AWS region to target.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.endpoint`
+
+Specify a custom endpoint for the AWS API.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials`
+
+Manually configure the AWS credentials to use (optional). For more information, see the xref:guides:cloud/aws.adoc[Amazon Web Services guide].
+
+*Type*: `object`
+
+=== `kafka.sasl[].aws.credentials.profile`
+
+The profile from `~/.aws/credentials` to use.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.id`
+
+The ID of credentials to use.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.secret`
+
+The secret for the AWS credentials in use.
+
+include::components:partial$secret_warning.adoc[]
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.token`
+
+The token for the AWS credentials in use. This is a required value for short-term credentials.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.from_ec2_role`
+
+Use the credentials of a host EC2 machine configured to assume an https://docs.aws.amazon.com/IAM/latest/UserGuide/id_roles_use_switch-role-ec2.html[IAM role associated with the instance^].
+
+*Type*: `bool`
+
+*Default*: `false`
+
+Requires version 4.2.0 or newer
+
+=== `kafka.sasl[].aws.credentials.role`
+
+The role ARN to assume.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.sasl[].aws.credentials.role_external_id`
+
+An external ID to use when assuming a role.
+
+*Type*: `string`
+
+*Default*: `""`
+
+=== `kafka.timestamp`
+
+Sets a timestamp for each message. Leave this field empty to use the current timestamp.
+
+This field supports xref:configuration:interpolation.adoc#bloblang-queries[interpolation functions].
+
+*Type*: `string`
+
+```yml
+# Examples
+timestamp: ${! timestamp_unix() }
+timestamp: ${! metadata("kafka_timestamp_unix") }
+```
+
+=== `disable_content_encryption`
```

Comment:
Missing field description for all fields from here to the end of the page. Could you supply or give a source for this?

Reply:
Added with help from Ockam.